crates/sui-core/src/authority.rs

```diff
@@ -760,14 +760,34 @@ impl AuthorityState {
         self.metrics.total_cert_attempts.inc();
 
         if self.is_fullnode() {
-            return Err(SuiError::GenericStorageError(
-                "cannot execute cert without effects on fullnode".into(),
-            ));
+            return Err(SuiError::ExecutionUnsupportedError(format!(
+                "Cannot execute certificate without effects on fullnode! {:?}",
+                certificate.digest()
+            )));
         }
 
-        if self.is_cert_awaiting_sequencing(certificate)? {
-            debug!("shared object cert has not been sequenced by narwhal");
-            return Err(SuiError::SharedObjectLockNotSetError);
+        if certificate
```

Comment:
We should do this check in authority server?
Also there is `is_system_tx` method you could call directly on certificate I think

Reply:
Sounds good.